### PR TITLE
Added specialization repository and UserSpecialization entity

### PR DESCRIPTION
**Pull request by TRybina132**

### CODE REVIEW

**Review by Strafe153 — APPROVED** [review]

File: DataAccess/Configurations/UserSpecializationConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            builder
+                .HasKey(userSpec =>
+                    new
+                    {
+                        userSpec.UserId,
+                        userSpec.SpecializationId
+                    });
+
+            builder.
```
Comment (Strafe153) [inline DataAccess/Configurations/UserSpecializationConfiguration.cs:19]:
```suggestion
            builder
```

File: DataAccess/Configurations/UserSpecializationConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                HasOne(userSpec => userSpec.User)
```
Comment (Strafe153) [inline DataAccess/Configurations/UserSpecializationConfiguration.cs:20]:
```suggestion
                .HasOne(userSpec => userSpec.User)
```

File: DataAccess/Configurations/UserSpecializationConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                .WithMany(user => user.UserSpecializations)
+                .HasForeignKey(userSpec => userSpec.UserId);
+
+            builder.HasOne(userSpec => userSpec.Specialization)
```
Comment (Strafe153) [inline DataAccess/Configurations/UserSpecializationConfiguration.cs:24]:
```suggestion
            builder
                .HasOne(userSpec => userSpec.Specialization)
```

**Review by FedorivRostik — CHANGES_REQUESTED** [review]
Delete extra usings at userSpecialization

**Review by Strafe153 — CHANGES_REQUESTED** [review]
Move dot and a command to a new line

**Review by Strafe153 — APPROVED** [review]

**Review by YuraRov — APPROVED** [review]

**Review by YuraRov — APPROVED** [review]

**Review by MaxKravchuk — APPROVED** [review]

**Review by FedorivRostik — APPROVED** [review]

**Pull request merged**